pkg/networkmanager/firewall-client.js

```diff
@@ -31,11 +32,19 @@ var firewall = {
      * fetchZoneInfos, the value */
     zones: {},
     activeZones: new Set(),
+    /* Zones predefined by firewalld, from untrusted to trusted */
+    predefinedZones: ['drop', 'block', 'public', 'external',
+        'dmz', 'work', 'home', 'internal', 'trusted'],
```

Comment:
It's a bit awkward to hardcode these -- aren't they subject to change or local customization? Can they be read from firewalld somewhere?

Reply:
Ah I see I didnt address this.

Now they're not subject to user customization. And are documented here https://firewalld.org/documentation/zone/predefined-zones.html

`/lib/firewalld/zones` contains non-user defined zones, but on fedora these also contain FedoraServer and FedoraWorkstation, which aren't documented zones. It's also not possible to retrieve this particular list from the dbus api.

So I'm not sure what's best here, sadly zone definition don't contain any metric for "trust", so sorting them runtime isn't possible either.